pkg/tasks/pip.go

```diff
@@ -29,35 +29,20 @@ func parserPip(config *taskConfig, task *Task) error {
 	task.header = strings.Join(files, ", ")
 
 	for _, file := range files {
-		task.addAction(&pipInstall{file: file})
+		task.
+			addActionWithBuilder(fmt.Sprintf("install %s", file),
+				func(ctx *context) error {
+					result := command(ctx, "pip", "install", "--require-virtualenv", "-r", file).
+						AddOutputFilter("already satisfied").Run()
+
+					if result.Error != nil {
+						return fmt.Errorf("Pip failed: %s", result.Error)
+					}
+					return nil
+				}).
+			addFileChangeCondition(file).
+			addFeatureChangeCondition("python")
```

Comment:
I would prefer something like:

```go
task.
         setup(fmt.Sprintf("install %s", file),
               func(ctx *context) error {
                     ...
               }).
         onFileChange(file).
         onFeatureChange("python")
```

Reply:
What do you think?

Reply:
`onFileChange` is good.
`setup` doesn't say that we are creating an action/step, right?
How would you declare a condition like `addConditionFunc`?